docs/2_Step_Types.md

```diff
@@ -0,0 +1,77 @@
+# Step Types
+
+There are several built-in types of steps:
+
+### Source Code Management(SCM)
+
+Source Code Management step is meant to be the first step of a pipeline. When you add a new pipeline, you are required to set up the Souce Code Management step.
+
+In SCM step, Choose a authenticated account( for how to do authorization please refer to [Source Code Management Integration](./3_Source_Code_management.md)) and select a repository, fill the **Branch** field to specify which git branch to use.
+
+If you enable the **Webhook** option, it means that the pipeline can be triggered by a push event webhook. For example if you are using Github here, a Github webhook will be automatically created and once a push happens in relevant repository & branch, Github will send a POST request to Rancher server and trigger a new run of the pipeline.
+
+### Build
+
+Build step is for building a docker image. There are normally two cases of building an image in continuous integration using docker. 
+
+1. Build a context image to be used in later tasks such as testing.
+2. Build an image as the artifact of the CI process, which can be used in later testing and deployment.
+
+You can use a Dockerfile in your source code repository or upload and configure one which is excluded from the repository. You can configure the **Build Path** and **Image Tag** of the built image. The image tag should be the full image name containing image registry prefix.
```

Comment:
1. You can upload a Dockerfile or use a existing Dockerfile in your source code repository.
2. You can configure the **Build Path** and **Image Tag** of the image we are going to build.